File: hourglass-sum/golang/hourglass_sum.go
package golang

func hourglassSum(arr [][]int32) int32 {
	var row, col, skip, sum int32
	var maxSum int32 = -63
	for row < 4 {
		for col < 4 {
			for i := row; i < row+3; i++ {
				for j := col; j < col+3; j++ {
					skip++
					if skip == 4 || skip == 6 {
						continue
					}
					sum += arr[i][j]
				}
			}

			if sum > maxSum {
				maxSum = sum
			}
			sum = 0
			skip = 0
			col++
		}
		col = 0
		row++
	}

	return maxSum
}

File: hourglass-sum/java/HourGlassSum.java
import java.util.List;

public class HourGlassSum {
    public static int hourglassSum(List<List<Integer>> arr) {
        int row = 0;
        int col = 0;
        int skip = 0;
        int sum = 0;
        int maxSum = -63;

        while (row < 4) {
            while (col < 4) {
                for (int i = row; i < row + 3; i++) {
                    for (int j = col; j < col + 3; j++) {
                        skip++;
                        if (skip == 4 || skip == 6) {
                            continue;
                        }
                        sum += arr.get(i).get(j);
                    }
                }

                if (sum > maxSum) {
                    maxSum = sum;
                }
                sum = 0;
                skip = 0;
                col++;
            }
            col = 0;
            row++;
        }

        return maxSum;
    }
}
